build_tools/benchmarks/common/benchmark_thresholds.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2021 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""A list of benchmarks and their similarity thresholds."""
+
+import re
+
+from dataclasses import dataclass
+from enum import Enum
+
+
+class ThresholdUnit(Enum):
+  PERCENTAGE = 0
+  VALUE = 1
+
+
+@dataclass
+class BenchmarkThreshold:
+  """Similarity threshold for benchmarks matching a regular expression."""
+  # A regular expression to match against the benchmark identifier.
+  regex: re.Pattern
+  # A threshold for computing the benchmark value average. Benchmark sample
+  # values from consecutive runs and within the given range will be considered
+  # as similar (with some noise). They will be used to compute the moving
+  # average. The number will be interpreted as a percentage. What value to set
+  # depends on the noise range of the particular benchmark.
+  threshold: int
+  unit: ThresholdUnit
+
+  def get_threshold_str(self):
+    """Returns a string representation of the threshold."""
+    if self.unit == ThresholdUnit.PERCENTAGE:
+      return f"{self.threshold}%"
+    return self.threshold
+
+
+# A list of benchmarks and their similarity thresholds.
+BENCHMARK_THRESHOLDS = [
```

Comment:
since there is a regex that matches all benchmarks last in the list, this comment could note that order matters:

```suggestion
# A list of benchmarks and their similarity thresholds.
# If multiple regexes match a single benchmark, the first match is used.
BENCHMARK_THRESHOLDS = [
```

Reply:
Yup!